Modules/_collectionsmodule.c

```diff
@@ -143,6 +143,40 @@ freeblock(block *b)
     }
 }
 
+static block *
+deque_get_block(dequeobject *deque, Py_ssize_t *index)
+{
+    Py_ssize_t i = *index, m = i, n;
+    block *b;
+
+    if (i == 0) {
```

Comment:
Have you considered not having these special cases for the first and last indices here? It would simplify the code, and it would also reduce branching so potentially even improve performance.

Reply:
Corrected in 6317c78